0002-data-collection-extension/README.md

```diff
@@ -1,96 +1,96 @@
+- Asynchronous record processing.
+- QoS levels to confirm message delivery and processing.
+- Separate response messages for processing confirmation.
+- Data type identification: distinct resource paths for different formats, `<format_specifier>` embedded into resource path.
+- Endpoint identification: `<endpoint_token>` embedded into resource path.
+- Minimum network usage: Upload data in batches.
+A batch is a number of records uploaded in one network packet.
+- Timestamp generation: on the server side, using network packet arrival time as a timestamp if no timestamp is provided by device.
 
 ## Use cases
 
 ### UC1
-Device shadow. The user is only interested in the current status of the endpoint parameters. The endpoint updates them periodically.
+Device shadow.
+The user only wants to know the current status of the endpoint parameters.
+The endpoint updates them periodically.
 
 ### UC2
-Historical data collection. The user is interested in storing all collected data with timestamps for further processing and visualizing historical trends.
+Historical data collection.
+The user wants to store all collected data with timestamps for further processing and for visualizing historical trends.
 
 ## Design
 
 ### Batch uploads
-To reduce network usage, all records are uploaded in _batches_. All records in a batch are processed as one and have a single response status.
+To reduce network usage, all records are uploaded in *batches*.
+All records in a batch are processed as one record and have a single response status.
 
-### Timestamps
-Due to the fact that different application might need timestamps in different formats and precision, data collection extension does not provide any special handling for timestamps. That is the responsibility of higher layers to interpret any field as a timestamp.
+### No timestamp handling
+Due to the fact that different applications might need timestamps in different formats and precision, Data Collection extension does not provide any special handling for timestamps.
+There is no special field for a timestamp — it is the responsibility of higher layers to interpret any field as a timestamp.
 
-#### Absent timestamp handling
-As there is no special field for a timestamp, that's higher level's issue how to handle absent timestamps.
-
-The recommended solution is to save server timestamp upon network message receiving and pass it along the parsed data, so upper layers can use that timestamp if needed.
+Recommended solution: save server timestamp upon receiving a network message and pass it along the parsed data, so the upper layers can use that timestamp if needed.
 
 ### Request/response
-The extension uses client-initiated request/response pattern defined in [1/KP](/0001-kaa-protocol). A response is sent if a bucket requires a processing confirmation.
+The extension uses client-initiated request/response pattern defined in [1/KP](/0001-kaa-protocol).
```

Comment:
Link to the [specific section](https://github.com/kaaproject/kaa-rfcs/tree/master/0001-kaa-protocol#requestresponse-pattern).